gap/semigroups/semirms.gi

```diff
@@ -881,16 +881,21 @@ function(R)
   return MagmaIsomorphismByFunctionsNC(R, S, iso, inv);
 end);
 
-InstallMethod(IsIdempotentGenerated, "for a Rees 0-matrix semigroup",
-[IsReesZeroMatrixSemigroup],
+InstallMethod(IsIdempotentGenerated, "for a Rees 0-matrix subsemigroup",
+[IsReesZeroMatrixSubsemigroup],
 function(R)
-  local RR;
-  if not IsConnectedDigraph(RZMSDigraph(R)) then
+  local U, N;
+  if not IsReesZeroMatrixSemigroup(R) then
+    TryNextMethod();
+  elif not IsConnectedDigraph(RZMSDigraph(R)) then
     return false;
   fi;
-  RR := Range(RZMSNormalization(R));
-  return Group(Filtered(MatrixEntries(RR), x -> x <> 0)) =
-    UnderlyingSemigroup(RR);
+  U := UnderlyingSemigroup(R);
+  if not IsGroupAsSemigroup(U) then
+    TryNextMethod();
```

Comment:
The next method is the usual one in `gap/attributes/properties` which (basically) generates the idempotent generated semigroup and compares it to the semigroup itself. It's the one I've made a couple of changes to below that you've commented on.